src/devices/cellular.c

```diff
@@ -0,0 +1,534 @@
+        return cell_info.signal;
+}
+
+const char* cell_get_subscriber_number()
+{
+        return (const char*) cell_info.number;
+}
+
+const char* cell_get_IMEI()
+{
+        return (const char*) cell_info.imei;
+}
+
+static bool get_methods(const enum cellular_modem modem,
+                        const struct cell_modem_methods **methods)
+{
+        const char *modem_name = NULL;
+        *methods = NULL;
+
+        switch(modem) {
+        case CELLULAR_MODEM_SIM900:
+                modem_name = "SIM900";
+                *methods = get_sim900_methods();
+                break;
+        case CELLULAR_MODEM_UBLOX_SARA_U280:
+                modem_name = "UBlox Sara U280";
+                *methods = get_sara_u280_methods();
+                break;
+        default:
+                break;
+        }
+
+        if (!*methods) {
+                pr_warning_int_msg("[cell] Unknown modem code: ", modem);
+                return false;
+        }
+
+        at_cfg = (*methods)->get_at_config();
+        pr_info_str_msg("[cell] Loading driver for ", modem_name);
+        return true;
+}
+
+
+/**
+ * Does auto-bauding.  This allows the chip to figure out what serial
+ * settings to use (default 8N1) along with what baud rate to use.  This
+ * should complete within 3 seconds according to spec.  Also ensures
+ * that the modem is still there.
+ *
+ * NOTE: We don't use conveinence methods here as echo may be on still.
+ */
+bool autobaud_modem(struct serial_buffer *sb, size_t tries,
+                    const size_t backoff_ms)
+{
+        for(; tries > 0; --tries) {
+                if (gsm_ping_modem(sb)) {
+                        pr_info("[cell] auto-baud successful\r\n");
+                        return true;
+                }
+
+                pr_debug("[cell] Unexpected reply. Backing off...\r\n");
+                delayMs(backoff_ms);
+        }
+
+        return false;
+}
+
+enum cellular_modem probe_cellular_manuf(struct serial_buffer *sb)
+{
+        const char *msgs[2];
+        const size_t msgs_len = ARRAY_LEN(msgs);
+
+        serial_buffer_reset(sb);
+        serial_buffer_append(sb, "AT+CGMI");
+        const size_t count = cellular_exec_cmd(sb, READ_TIMEOUT, msgs, msgs_len);
+
+        if (!is_rsp_ok(msgs, count))
+                return CELLULAR_MODEM_UNKNOWN;
+
+        pr_info_str_msg("[cell] manufacturer is ", msgs[0]);
+
+        if (strstr(msgs[0], "u-blox"))
+                return CELLULAR_MODEM_UBLOX_SARA_U280;
+        else if (strstr(msgs[0], "SIMCOM_Ltd"))
+                return CELLULAR_MODEM_SIM900;
+        else
+                return CELLULAR_MODEM_UNKNOWN;
+}
+
+telemetry_status_t cellular_get_connection_status()
+{
+        return telemetry_info.status;
+}
+
+int32_t cellular_active_time()
+{
+        const tiny_millis_t since = telemetry_info.active_since;
+        return 0 == since ? 0 : (int) getUptime() - since;
+}
+
+const char* cellular_get_net_status_desc()
+{
+        switch(cell_info.net_status) {
```

Comment:
The other status codes are returned as numbers and the app decodes it. I think we should remain consistent with this and let the app convert that into textual strings.